webapp/src/RepoGit.ts

```diff
@@ -67,7 +69,14 @@ export class RepoGit {
    */
   public async checkoutBaseAndPull(): Promise<string> {
     await this.git.checkout(this.spConfig.baseBranch);
-    await this.git.pull();
+
+    const now = new Date();
+    const age = now.getTime() - this.lastPullTime.getTime();
+    if (age > 30000) {
+      // We only pull if old
+      await this.git.pull();
+      this.lastPullTime = now;
+    }
```

Comment:
Once age is older than 30 s, concurrent requests will race this `pull` as `lastPullTime` is only updated after any of them completes. If we update `lastPullTime` before we await `pull`, only one of the concurrent requests will find the age old and there will be less concurrent pulling. We can fix that in a separate PR.